src/include/storage/data_table.h

```diff
@@ -223,8 +223,12 @@ class DataTable {
  private:
   // The ArrowSerializer needs access to its blocks.
   friend class ArrowSerializer;
-  // The GarbageCollector needs to modify VersionPtrs when pruning version chains
+  // The TransactionContext needs to modify VersionPtrs when pruning version chains
+  friend class transaction::TransactionContext;
+  // TODO(Ling): remove this after deferred action is installed
```

Comment:
Is this TODO still relevant?

Reply:
Yes, but it means "remove this after gc is removed in the future". I will fix the todo comment